### PR TITLE
[SPARK-28764][CORE][TEST] Remove writePartitionedFile in ExternalSorter

### PR DESCRIPTION
**Pull request by smallzhongfeng**
### What changes were proposed in this pull request?

Remove `writePartitionedFile` as this is only used by UnsafeRowSerializerSuite in the SQL project.

### Why are the changes needed?

Remove `writePartitionedFile` in `ExternalSorter` and update `UnsafeRowSerializerSuite`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Update ut.

### CODE REVIEW

**Comment by smallzhongfeng** [comment]
cc @mccheah @cloud-fan @HyukjinKwon If I have misunderstood, thank you very much for pointing it out :).

**Comment by LuciferYang** [comment]
also cc @Ngone51 @mridulm FYI

**Comment by mridulm** [comment]
We are not cleaning up the file now, right ?

**Review by cloud-fan — APPROVED** [review]

**Comment by smallzhongfeng** [comment]
> We are not cleaning up the file now, right ?

Thanks for your review. @mridulm 

You can see https://github.com/apache/spark/blob/93f3384d8014bd8a92a3631b7e817f1d6817c11e/core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala#L368. When sparkEnv stops, the diskBlockManager will stop immediately and the temporary folder will be deleted.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeRowSerializerSuite.scala [hunk 1]
```diff
@@ -125,7 +130,7 @@ class UnsafeRowSerializerSuite extends SparkFunSuite with LocalSparkSession {
     assert(sorter.numSpills > 0)
 
     // Merging spilled files should not throw assertion error
-    sorter.writePartitionedFile(ShuffleBlockId(0, 0, 0), outputFile)
+    sorter.writePartitionedMapOutput(0, 0, mapOutputWriter)
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeRowSerializerSuite.scala:133]: I remember we remove local directories when the JVM (for testing) is shut down. But would be great to confirm that before we merge this.

Comment (HyukjinKwon) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeRowSerializerSuite.scala:133]: Oh, just read https://github.com/apache/spark/pull/39368?notification_referrer_id=NT_kwDOAGLXhbI1MjMzNjcxNTQ0OjY0Nzc3MDE&notifications_query=is%3Aunread#issuecomment-1373501036. LGTM too

**Review by Ngone51 — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Review by mridulm — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**